examples/random.h

```diff
@@ -0,0 +1,54 @@
+/**********************************************************************
+ * Copyright (c) 2020 Elichai Turkel	                              *
+ * Distributed under the MIT software license, see the accompanying   *
+ * file COPYING or http://www.opensource.org/licenses/mit-license.php.*
+ **********************************************************************/
+/*
+ * In this file only the best practice randomness sources are used
+ * Platform randomness sources:
+ * Linux   -> `getrandom(2)`(`sys/random.h`), if not available `/dev/urandom` should be used. http://man7.org/linux/man-pages/man2/getrandom.2.html, https://linux.die.net/man/4/urandom
+ * macOS   -> `getentropy(2)`(`sys/random.h`), if not available `/dev/urandom` should be used. https://www.unix.com/man-page/mojave/2/getentropy, https://opensource.apple.com/source/xnu/xnu-517.12.7/bsd/man/man4/random.4.auto.html
+ * FreeBSD -> `getrandom(2)`(`sys/random.h`), if not available `kern.arandom` should be used. https://www.freebsd.org/cgi/man.cgi?query=getrandom, https://www.freebsd.org/cgi/man.cgi?query=random&sektion=4
+ * OpenBSD -> `getentropy(2)`(`unistd.h`), if not available `/dev/urandom` should be used. https://man.openbsd.org/getentropy, https://man.openbsd.org/urandom
```

Comment:
> I think the reason was that some platforms still use RC4 for that syscall:

  Urgh. But we're only targeting `APPLE` and `OpenBSD` here, whose manpages both recommend arc4random. But agree that `getentropy` should be ok too.

  How about we replace the sentence "In this file only the best practice randomness sources are used" at the top of the file with:
  "This file is an attempt at collecting best practice methods for obtaining randomness with different operating systems. It may be out-of-date. Consult the documentation of the operating system before considering to use the methods below."

Reply:
That sounds good. I hate that OSes are so bad that no one can actually tell what it actually the best practice

Reply:
nit: better formatting
```
 * This file is an attempt at collecting best practice methods for obtaining
 * randomness with different operating systems. It may be out-of-date. Consult
 * the documentation of the operating system before considering to use the
 * methods below.
 *
 * Platform randomness sources:
 ```